pkg/engine/internal/planner/physical/printer.go

```diff
@@ -27,6 +27,8 @@ func toTree(p *Plan, n Node) *tree.Node {
 
 func toTreeNode(n Node) *tree.Node {
 	treeNode := tree.NewNode(n.Type().String(), "")
+	treeNode.Context = n
```

Comment:
This is used by `workflow.Sprint` to allow workflow printing to hook into the tree produced by the physical plan and add additional context for nodes (the streams to write to or read from).